File: compiler_v2/src/main.cc
#include <iostream>
#include "lexer.h"

std::string read() {
  return R"(
  i8 x = 4;
  mut i8 y = x;
)";
}

int main() {
  std::string in = read();
  auto y = lexer(in);
  for (auto x : y) {
    std::cout << (int)x.type << ' ' << x.val << '\n';
  }
}

File: compiler_v2/src/main.zig
const std = @import("std");
const l = @import("lexer.zig");

pub fn main() !void {
    var gpa = std.heap.GeneralPurposeAllocator(.{}){};
    const alloc = gpa.allocator();
    defer _ = gpa.deinit();
    std.debug.print("Token len {d}\n", .{l.lex(alloc, "u8 x = 5;").items.len});
}
